services/QuillLMS/app/models/concerns/public_progress_reports.rb

```diff
@@ -319,29 +320,49 @@ def get_previously_assigned_recommendations_by_classroom(classroom_id, activity_
     classroom = Classroom.find(classroom_id)
     teacher_id = classroom.owner.id
     diagnostic = Activity.find(activity_id)
-    assigned_recommendations = RecommendationsQuery.new(diagnostic.id).activity_recommendations.map do |rec|
+
+    release_method = ActivityPackSequence.find_by(classroom: classroom, diagnostic_activity: diagnostic)&.release_method
+
+    assigned_recommendations = RecommendationsQuery.new(diagnostic.id).activity_recommendations.map do |recommendation|
+      unit_template_id = recommendation[:activityPackId]
       # teachers may rename and reassign activity packs so we check all
-      units = Unit.where(user_id: teacher_id, unit_template_id: rec[:activityPackId], visible: true)
+      units = Unit.where(user_id: teacher_id, unit_template_id: unit_template_id, visible: true)
       if !units
-        units = Unit.where(user_id: teacher_id, name: UnitTemplate.find_by_id(rec[:activityPackId]).name, visible: true)
+        name = UnitTemplate.find_by(id: unit_template_id).name
+        units = Unit.where(user_id: teacher_id, name: name, visible: true)
       end
-      student_ids = []
-      units.each do |unit|
-        student_ids.concat(ClassroomUnit.find_by(unit: unit, classroom: classroom, visible: true).try(:assigned_student_ids) || [])
-      end
-      return_value_for_recommendation(student_ids.uniq, rec)
+
+      student_ids =
+        classroom
+          .classroom_units
+          .visible
+          .where(unit: units)
+          .pluck(:assigned_student_ids)
+          .flatten
+          .uniq
+
+      return_value_for_recommendation(student_ids, recommendation)
     end
-    recommended_lesson_activity_ids = LessonRecommendationsQuery.new(diagnostic.id)
-      .activity_recommendations
-      .map { |r| r[:activityPackId] }
+
+    recommended_lesson_activity_ids =
+      LessonRecommendationsQuery
+        .new(diagnostic.id)
+        .activity_recommendations
+        .pluck(:activityPackId)
+
     associated_teacher_ids = ClassroomsTeacher.where(classroom_id: classroom_id).pluck(:user_id)
-    assigned_lesson_ids = Unit.where(unit_template_id: recommended_lesson_activity_ids, user_id: associated_teacher_ids)
-      .joins(:classroom_units)
-      .where("classroom_units.classroom_id=?", classroom_id)
-      .pluck(:unit_template_id)
+
+    assigned_lesson_ids =
+      Unit
+        .where(unit_template_id: recommended_lesson_activity_ids, user_id: associated_teacher_ids)
+        .joins(:classroom_units)
+        .where(classroom_units: { classroom_id: classroom_id })
+        .pluck(:unit_template_id)
+
     {
       previouslyAssignedIndependentRecommendations: assigned_recommendations,
-      previouslyAssignedLessonsRecommendations: assigned_lesson_ids
+      previouslyAssignedLessonsRecommendations: assigned_lesson_ids,
+      releaseMethod: release_method
```

Comment:
I might be missing something, and it doesn't look like it's documented in Notion (unfortunately) but I think we had talked in a huddle and on Slack (last reference [here](https://quill.slack.com/archives/D01LD5AUQH2/p1661359033026189)) about having `releaseMethod` send up either `immediate` or `staggered` if the recommendations had already been assigned, or `null` if it hadn't. I think the way you have it here, it will always either send up `staggered` or `null`. I can update the frontend to make it work with this setup if you'd prefer, but I wanted to call that out.